### PR TITLE
Enable mTLS cluster-wide

### PR DESCRIPTION
**Pull request by kubadz**
**Description**

Changes proposed in this pull request:

- Enable mTLS cluster-wide

**Related issue(s)**
See #3320

### CODE REVIEW

File: components/istio-kyma-patch/patch.sh [hunk 1]
```diff
@@ -26,10 +26,11 @@ function require_istio_system() {
     kubectl get namespace istio-system >/dev/null
 }
 
-function require_mtls_disabled() {
```
Comment (kubadz) [inline components/istio-kyma-patch/patch.sh]: TODO: move changes in istio-kyma-patch to another PR

File: resources/event-bus/charts/subscription-controller-knative/templates/deployment.yaml [hunk 2]
```diff
@@ -32,15 +36,19 @@ spec:
             - name: http
               containerPort: {{ .Values.port }}
           livenessProbe:
-            httpGet:
-              path: /v1/status/live
-              port: http
+            exec:
+              command:
+              - curl
```
Comment (radufa) [inline resources/event-bus/charts/subscription-controller-knative/templates/deployment.yaml]: "curl" is not found: 
```
Events:
  Type     Reason     Age                 From               Message
  ----     ------     ----                ----               -------
  Normal   Scheduled  35m                 default-scheduler  Successfully assigned kyma-system/event-bus-subscription-controller-knative-8b8b8755f-llkgd to minikube
  Normal   Pulling    35m                 kubelet, minikube  pulling image "eu.gcr.io/kyma-project/develop/event-bus-subscription-controller-knative:24302b4a"
  Normal   Pulled     35m                 kubelet, minikube  Successfully pulled image "eu.gcr.io/kyma-project/develop/event-bus-subscription-controller-knative:24302b4a"
  Normal   Created    32m (x3 over 35m)   kubelet, minikube  Created container
  Normal   Started    32m (x3 over 35m)   kubelet, minikube  Started container
  Normal   Pulled     32m (x2 over 33m)   kubelet, minikube  Container image "eu.gcr.io/kyma-project/develop/event-bus-subscription-controller-knative:24302b4a" already present on machine
  Normal   Killing    30m (x4 over 33m)   kubelet, minikube  Killing container with id docker://subscription-controller-knative:Container failed liveness probe.. Container will be killed and recreated.
  Warning  Unhealthy  25m (x19 over 34m)  kubelet, minikube  Readiness probe failed: OCI runtime exec failed: exec failed: container_linux.go:344: starting container process caused "exec: \"curl\": executable file not found in $PATH": unknown
  Warning  Unhealthy  14m (x25 over 34m)  kubelet, minikube  Liveness probe failed: OCI runtime exec failed: exec failed: container_linux.go:344: starting container process caused "exec: \"curl\": executable file not found in $PATH": unknown
  Warning  BackOff    4m (x77 over 27m)   kubelet, minikube  Back-off restarting failed container
```

File: resources/core/charts/kubeless/templates/tests/test-kubeless.yaml [hunk 3]
```diff
@@ -62,9 +62,10 @@ metadata:
   labels:
     helm-chart-test: "true"
   annotations:
-    sidecar.istio.io/inject: "false"
+    sidecar.istio.io/inject: "true"
```
Comment (jakkab) [inline resources/core/charts/kubeless/templates/tests/test-kubeless.yaml]:
```suggestion
    sidecar.istio.io/inject: true
```

File: resources/core/charts/kubeless/templates/tests/test-kubeless.yaml [hunk 4]
```diff
@@ -62,9 +62,10 @@ metadata:
   labels:
     helm-chart-test: "true"
   annotations:
-    sidecar.istio.io/inject: "false"
+    sidecar.istio.io/inject: true
```
Comment (jakkab) [inline resources/core/charts/kubeless/templates/tests/test-kubeless.yaml]:
```suggestion
    sidecar.istio.io/inject: "true"
```

File: resources/application-connector/values.yaml [hunk 5]
```diff
@@ -16,11 +16,11 @@ global:
...
   application_operator:
-    dir: develop/
-    version: ee38fb8f
+    dir: pr/
+    version: PR-3666
   application_operator_tests:
-    dir: develop/
-    version: 2b4eed1c
+    dir: pr/
+    version: PR-3666
```
Comment (akucharska) [inline resources/application-connector/values.yaml]: Remember to change it :)

File: resources/istio-kyma-patch/README.md [hunk 6]
```diff
@@ -9,4 +9,5 @@ By default, following changes are applied:
  * Configuration of the [sidecar injector](../../components/istio-kyma-patch/README.md).
  * The Egressgateway, Ingressgateway, policy, and telemetry are configured to use Zipkin from the `kyma-system` Namespace.
  * Monitoring and tracing related resources are deleted.
- * Sidecar injection is enabled in all Namespaces, except those labeled with `istio-injection: disabled`.
\ No newline at end of file
+ * Sidecar injection is enabled in all Namespaces, except those labeled with `istio-injection: disabled`.
+ * A DestinationRule CR disabling (m)TLS for requests to helm tiller.
```
Comment (klaudiagrz) [inline resources/istio-kyma-patch/README.md]:
```suggestion
 * A DestinationRule CR disabling mTLS for requests to Helm Tiller.
```

**Comment by sayanh** [comment]
Can you please add changes for this file https://github.com/kyma-project/kyma/blob/09fcd504d0dcb8a4d746d7a321908b89a4c508e5/tests/end-to-end/backup-restore-test/deploy/chart/backup-test/templates/test/test-backup.yaml as well?

**Comment by sayanh** [comment]
Can you please enable side-car here https://github.com/kyma-project/kyma/blob/09fcd504d0dcb8a4d746d7a321908b89a4c508e5/tests/end-to-end/upgrade/chart/upgrade/templates/test/test-upgrade.yaml?

File: tests/end-to-end/backup-restore-test/deploy/chart/backup-test/templates/test/test-backup.yaml [hunk 7]
```diff
@@ -104,5 +105,23 @@ spec:
       value: "/all-backup.yaml"
     - name: SYSTEM_BACKUP_CONFIGURATION_FILE
       value: "/system-backup.yaml"
+  - image: eu.gcr.io/kyma-project/test-infra/alpine-kubectl:v20190325-ff66a3a
+    name: manager
+    command:
+      - bin/sh
+      - -c
+      - |
+        SERVICE="e2e-upgrade-test"
```
Comment (sayanh) [inline tests/end-to-end/backup-restore-test/deploy/chart/backup-test/templates/test/test-backup.yaml]:
```suggestion
        SERVICE="restore.test"
```

File: resources/core/charts/kubeless/templates/tests/test-kubeless.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (Demonsthere) [reply resources/core/charts/kubeless/templates/tests/test-kubeless.yaml]: We are aware of that, yet people prefer to have it explicitly written, so we keep it

File: tests/end-to-end/backup-restore-test/deploy/chart/backup-test/templates/test/test-backup.yaml [hunk 8]
```diff
@@ -86,8 +86,9 @@ metadata:
     heritage: "{{ .Release.Service }}"
   annotations:
     "helm.sh/hook": test-success
-    "sidecar.istio.io/inject": "false"
+    "sidecar.istio.io/inject": "true"
 spec:
+  shareProcessNamespace: true
   serviceAccount: {{ template "backup-test.fullname" . }}
   containers:
   - name: backup-test
```
Comment (sayanh) [inline tests/end-to-end/backup-restore-test/deploy/chart/backup-test/templates/test/test-backup.yaml]:
```
    - name: backup-test
      command:
        args:
        - /bin/sh
        - -c
        - |
          sleep 10

          while ! curl -Ik https://$KUBERNETES_SERVICE_HOST:$KUBERNETES_SERVICE_PORT_HTTPS >/dev/null 2>/dev/null
          do
              sleep 1
          done

          /restore.test -test.v
          exit_code=$?

          pkill -TERM pilot-agent
          sleep 4

          exit $exit_code
```

File: tests/end-to-end/backup-restore-test/deploy/chart/backup-test/templates/test/test-backup.yaml [hunk 9]
```diff
@@ -104,5 +105,23 @@ spec:
       value: "/all-backup.yaml"
     - name: SYSTEM_BACKUP_CONFIGURATION_FILE
       value: "/system-backup.yaml"
+  - image: eu.gcr.io/kyma-project/test-infra/alpine-kubectl:v20190325-ff66a3a
```
Comment (sayanh) [inline tests/end-to-end/backup-restore-test/deploy/chart/backup-test/templates/test/test-backup.yaml]: Lets get rid of this container completely

File: tests/end-to-end/upgrade/chart/upgrade/templates/test/test-upgrade.yaml [hunk 10]
```diff
@@ -23,4 +24,22 @@ spec:
       value: "info"
     - name: DOMAIN
       value: "{{ .Values.global.domainName }}"
+  - image: eu.gcr.io/kyma-project/test-infra/alpine-kubectl:v20190325-ff66a3a
```
Comment (sayanh) [inline tests/end-to-end/upgrade/chart/upgrade/templates/test/test-upgrade.yaml]: Here, this approach is a bit dangerous. Incase the test pod is slow in ContainerCreation process, we will kill istio-proxy. When the test pod runs the tests, everything will fail.
Why don't we apply this approach https://github.com/kyma-project/kyma/pull/3385/files#diff-da0dbb74a6bb4540d5dc5fadfa5913c9R54?

**Pull request merged**